nxcomp/Loop.cpp

```diff
@@ -9340,6 +8631,100 @@ int ParseCommandLineOptions(int argc, const char **argv)
 
           return -1;
         }
+        case 'd':
+        {
+          if ( argi+1 >= argc )
+          {
+            PrintUsageInfo(nextArg, 0);  
+            return -1;
+          }
+
+          int level = atoi(argv[argi+1]);
```

Comment:
`atoi` has undefined behavior if its argument is outside of the convertible range.

Please check my code in https://github.com/ArcticaProject/nx-libs/blob/3.6.x/nx-X11/programs/Xserver/hw/nxagent/Args.c#L1370 for a (hopefully) more robust alternative.